boltdb.go

```diff
@@ -49,8 +49,11 @@ func NewBoltDBWithOpts(name string, dir string, opts *bbolt.Options) (DB, error)
 	dbPath := filepath.Join(dir, name+".db")
 	db, err := bbolt.Open(dbPath, os.ModePerm, opts)
 	if err != nil {
-		return nil, err
-	}
+		err = os.MkdirAll(dbPath, 0755)
```

Comment:
The target still needs to be a file, so if we create the whole path it will still fail. What I think we probably want here is something like:

```go
if err := os.MkdirAll(filepath.Dir(dbPath), 00700); err != nil {
   return nil, err
}
db, err := bbolt.Open(dbPath, os.ModePerm, opts)
// ...
```

or words to that effect (key point being to only create the parent)

Reply:
I'll test with a gaia state sync...